### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6240

## Notes
The issue we faced earlier with docker server not running for the test module was fixed with following lines.
```groovy
test.mustRunAfter startNatsServersForTests
```

```groovy
task ballerinaTest {
    dependsOn(":${packageName}-ballerina:build")
    dependsOn(updateTomlFiles)
    dependsOn(startNatsServersForTests) // Adding this line
    dependsOn(initializeVariables)
    finalizedBy(commitTomlFiles)
    
    // Remaining task configuration...
}

```

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina-tests/build.gradle [hunk 1]
```diff
@@ -182,6 +153,36 @@ task stopNatsServersForTests() {
...
+            }
+        }
+    }
+}
+
+test {
+    dependsOn(ballerinaTest)
+}
+
+test.mustRunAfter startNatsServersForTests
```
Comment (dilanSachi) [inline ballerina-tests/build.gradle:180]: do we need both this and `dependsOn(startNatsServersForTests)` in L159 above?

File: .github/workflows/trigger-load-tests.yml [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
+            clusterName:
+                description: 'Cluster name'
+                default: 'nats-perf-cluster-test'
+                required: false
+            branch:
+                description: 'Branch of the given repository'
+                default: ''
+                required: false
+    schedule:
+        -   cron: '0 20 * * *'
```
Comment (dilanSachi) [inline .github/workflows/trigger-load-tests.yml:19]:
```suggestion
        -   cron: '30 19 * * *'
```

File: .github/workflows/build-with-ballerina-lang.yml
Comment (ThisaruGuruge) [inline .github/workflows/build-with-ballerina-lang.yml:1]: What's the purpose of this workflow?

File: .github/workflows/daily-build.yml [hunk 3]
```diff
@@ -17,12 +17,12 @@ jobs:
...
           distribution: 'temurin'
           java-version: 17.0.7
-
+      
       - name: Set environment variable
         if: github.event.action == 'check_connector_for_breaking_changes'
         run: |
           echo "BUILD_USING_DOCKER=-PbuildUsingDocker=nightly" >> $GITHUB_ENV
-          echo "GRADLE_SKIP_TASKS=-x :nats-compiler-plugin-tests:test -x :nats-ballerina-tests:build -x :nats-ballerina-tests:test" >> $GITHUB_ENV
+          echo "GRADLE_SKIP_TASKS=-x :nats-compiler-plugin-tests:test"  >> $GITHUB_ENV
```
Comment (ThisaruGuruge) [inline .github/workflows/daily-build.yml:25]: Do we need these values to be set as env.s here? Can be directly applied in the next step as it seems. (Also, any specific reason to disable the compiler plugin tests?)

File: .github/workflows/daily-build.yml [hunk 4]
```diff
@@ -17,12 +17,11 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: 17.0.7
-
+      
       - name: Set environment variable
         if: github.event.action == 'check_connector_for_breaking_changes'
         run: |
           echo "BUILD_USING_DOCKER=-PbuildUsingDocker=nightly" >> $GITHUB_ENV
-          echo "GRADLE_SKIP_TASKS=-x :nats-compiler-plugin-tests:test -x :nats-ballerina-tests:build -x :nats-ballerina-tests:test" >> $GITHUB_ENV
```
Comment (ThisaruGuruge) [inline .github/workflows/daily-build.yml:25]: We can remove this step.

File: .github/workflows/daily-build.yml [hunk 5]
```diff
@@ -32,6 +31,7 @@ jobs:
         run: |
           ./gradlew clean build $GRADLE_SKIP_TASKS $BUILD_USING_DOCKER
```
Comment (ThisaruGuruge) [inline .github/workflows/daily-build.yml:32]: After removing the above step (Set environment variable), we can directly put the Gradle build options here. 

```suggestion
          ./gradlew clean build -P buildUsingDocker=nightly
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**